src/bson.c

```diff
@@ -862,8 +862,8 @@ void object_to_bson(zval *object, php_phongo_bson_flags_t flags, const char *key
 			tmp_ht = HASH_OF(obj_data);
 #endif
 
-			if (tmp_ht) {
-				ZEND_HASH_APPLY_COUNT(tmp_ht)++;
+			if (tmp_ht && ZEND_HASH_APPLY_PROTECTION(tmp_ht)) {
+				ZEND_HASH_INC_APPLY_COUNT(tmp_ht);
```

Comment:
If we look at PHP's `json_encode()`, the apply count is always incremented when iterating on arrays and objects (compare [PHP-5.6](https://github.com/php/php-src/blob/PHP-5.6/ext/json/json.c#L253) and [PHP-7.0](https://github.com/php/php-src/blob/PHP-7.0/ext/json/json_encoder.c#L131)).

The `ZEND_HASH_APPLY_PROTECTION()` macros back to @dstogov's original commit in https://github.com/php/php-src/commit/b3b616cf7ee0702aca2f6fdb022dc7e28e015a22.